certora/specs/bridge.spec

```diff
@@ -500,27 +504,60 @@ function callFunctionSetParams(
     method f, env e, address receiver,
     address aToken, address asset,
     uint256 amount, bool fromToUnderlyingAsset) returns uint256 {
+    // Inhibits the user from calling the functions withdraw and receiveRewards.
+    // Expect unreachability for these functions (intended). 
     requireInvariant alwaysUnSent(e);
     if (f.selector == initiateWithdraw_L2(address, uint256, address, bool).selector){
+        require receiver != currentContract;
         return initiateWithdraw_L2(e, aToken, amount, receiver, fromToUnderlyingAsset); 
     }   
     else if (f.selector == deposit(address, uint256, uint256, uint16, bool).selector){
         uint256 l2Recipient = BRIDGE_L2.address2uint256(receiver);
         uint16 referralCode;
+        require receiver != currentContract;
         return deposit(e, aToken, l2Recipient, amount, referralCode, fromToUnderlyingAsset);
     }
     else if (f.selector == bridgeRewards_L2(address, uint256).selector) {
         bridgeRewards_L2(e, receiver, amount);
         return 0;
     }
-    else if (f.selector == withdraw(address, uint256, address, uint256, uint256, bool).selector) {
-        uint256 l2sender;
-        withdraw(e, aToken, l2sender, receiver, amount, BRIDGE_L2.l2RewardsIndex(), fromToUnderlyingAsset);
-        return 0;
-    }
     else {
         calldataarg args;
         f(e, args);
         return 0;
     }     
-}
\ No newline at end of file
+}
+
+////////////////////////////////////////////////////////////////////////////
+//                       Summarizations                                   //
+////////////////////////////////////////////////////////////////////////////
+/*
+The following functions are used as summarization (under-approximation)
+for the real functions in the code rayMul and rayDiv.
```

Comment:
this is equivalent to assuming that the index is 2, right? you can explain this so it will give the incentive for this.
One can instead of this use constantL1Index for specific rules, right?

Reply:
The summarization allows any constant value for the index, not just 2. But it is a global requirement, i.e. for every call to this function, for every asset.
The function constantL1Index is asset dependent (and can be triggered on/off).
Only with the summarization one can reduce the run times.